jobapi/routes.go

```diff
@@ -0,0 +1,148 @@
+package jobapi
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/buildkite/agent/v3/agent"
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/chi/v5/middleware"
+	"golang.org/x/exp/maps"
+)
+
+var protectedEnv map[string]struct{}
+
+func init() {
+	protectedEnv = make(map[string]struct{}, len(agent.ProtectedEnv))
+	for _, p := range agent.ProtectedEnv {
+		protectedEnv[p] = struct{}{}
+	}
+}
+
+func (s *Server) router() chi.Router {
+	r := chi.NewRouter()
+	r.Use(
+		middleware.Recoverer,
+		// middleware.Logger, // REVIEW: Should we log requests to this API? If so, where should we log them to? The job logs?
+		HeadersMdlw(),
+		AuthMdlw(s.token),
+	)
+
+	r.Route("/api/current-job/v0", func(r chi.Router) {
+		r.Get("/env", s.getEnv())
+		r.Patch("/env", s.patchEnv())
+		r.Delete("/env", s.deleteEnv())
+	})
+
+	return r
+}
+
+func (s *Server) getEnv() http.HandlerFunc {
```

Comment:
What about:

```go
r.Route(..., func(r chi.Router) {
    r.Get("/env", s.getEnv)   // no parentheses on s.getEnv
    ...
})

func (s *Server) getEnv(w http.ResponseWriter, _ *http.Request) {
     // no func generator needed
}

Reply:
oh yeah, good call. i think i used the func generator pattern for the middlewares because some of them actually need stuff passed into them and then blindly used them for the route functions too. 

will update